File: app/schemas/task.py
"""
任务相关的 Pydantic Schema
"""

from datetime import datetime
from typing import Optional, Dict, Any, List

from pydantic import BaseModel, Field


class TaskHistoryResponse(BaseModel):
    """任务历史响应"""
    id: int
    task_id: str
    user_id: Optional[int] = None
    task_type: str
    prompt: str
    negative_prompt: Optional[str] = None
    parameters: Dict[str, Any] = {}
    status: str
    result_path: Optional[str] = None
    result_filename: Optional[str] = None
    error_message: Optional[str] = None
    created_at: Optional[datetime] = None
    started_at: Optional[datetime] = None
    completed_at: Optional[datetime] = None
    execution_time: Optional[float] = None
    
    model_config = {"from_attributes": True}


class TaskHistoryListResponse(BaseModel):
    """任务历史列表响应"""
    items: List[TaskHistoryResponse]
    total: int
    page: int
    page_size: int
    total_pages: int


class TaskStatistics(BaseModel):
    """任务统计"""
    total_tasks: int = Field(..., description="总任务数")
    completed_tasks: int = Field(..., description="完成任务数")
    failed_tasks: int = Field(..., description="失败任务数")
    pending_tasks: int = Field(..., description="等待中任务数")
    
    # 按类型统计
    text_to_image_count: int = Field(0, description="文生图任务数")
    image_edit_count: int = Field(0, description="图像编辑任务数")
    batch_edit_count: int = Field(0, description="批量编辑任务数")
    
    # 时间统计
    avg_execution_time: Optional[float] = Field(None, description="平均执行时间（秒）")
    total_execution_time: Optional[float] = Field(None, description="总执行时间（秒）")


class UserQuotaResponse(BaseModel):
    """用户配额响应"""
    user_id: int
    daily_limit: int = Field(..., description="每日限额")
    monthly_limit: int = Field(..., description="每月限额")
    used_today: int = Field(..., description="今日已用")
    used_this_month: int = Field(..., description="本月已用")
    total_used: int = Field(..., description="总使用量")
    remaining_today: int = Field(..., description="今日剩余")
    remaining_this_month: int = Field(..., description="本月剩余")
    
    model_config = {"from_attributes": True}

File: frontend/src/types/task.ts
export type TaskStatus = 'pending' | 'running' | 'completed' | 'failed' | 'cancelled'

export type TaskType = 'text_to_image' | 'image_edit' | 'batch_edit'

export interface Task {
  task_id: string
  status: TaskStatus
  error: string | null
  created_at: string
  started_at: string | null
  completed_at: string | null
  position_in_queue: number
  wait_time_seconds: number | null
  execution_time_seconds: number | null
  task_type?: TaskType
  result_path?: string
  result_filename?: string
}

export interface TaskSubmitResponse {
  message: string
  task_id: string
  status_url: string
  result_url: string
  queue_info: {
    pending_tasks: number
    running_tasks: number
  }
}

export interface TaskHistory {
  id: number
  task_id: string
  user_id: number | null
  task_type: TaskType
  prompt: string
  negative_prompt: string | null
  parameters: Record<string, unknown>
  status: TaskStatus
  result_path: string | null
  result_filename: string | null
  error_message: string | null
  created_at: string
  started_at: string | null
  completed_at: string | null
  execution_time: number | null
}

export interface TaskHistoryListResponse {
  items: TaskHistory[]
  total: number
  page: number
  page_size: number
  total_pages: number
}

export interface TaskStatistics {
  total_tasks: number
  completed_tasks: number
  failed_tasks: number
  pending_tasks: number
  text_to_image_count: number
  image_edit_count: number
  batch_edit_count: number
  avg_execution_time: number | null
  total_execution_time: number | null
  today_tasks?: number
  active_users?: number
}

export interface QueueInfo {
  is_running: boolean
  gpu_count: number
  max_workers: number
  queue_size: number
  tasks: {
    pending: number
    running: number
    completed: number
    failed: number
    total: number
  }
}

export interface UserQuota {
  user_id: number
  daily_limit: number
  monthly_limit: number
  used_today: number
  used_this_month: number
  total_used: number
  remaining_today: number
  remaining_this_month: number
}
